pennylane/estimator/resource_operator.py

```diff
@@ -306,12 +338,26 @@ def pow_resource_decomp(
         r"""Returns a list representing the resources for an operator
         raised to a power.
 
+        For ResourceOperators that don't define a `pow_resource_decomp`, this will be its
+        `pow_resource_decomp` method. By default, this method returns the operator multiplied
+        by the given power.
+
         Args:
             pow_z (int): exponent that the operator is being raised to
             target_resource_params (dict | None): A dictionary containing the resource parameters
                 of the target operator.
         """
-        raise ResourcesUndefinedError
+        if pow_z == 0:
+            return [GateCount(resource_rep(qre_ops.Identity))]
+
+        target_resource_params = target_resource_params or {}
+        num_wires = target_resource_params.get("num_wires")
+        base_cmpr_op = CompressedResourceOp(cls, num_wires, target_resource_params)
```

Comment:
This shouldn't work in general because `num_wires` isn't always a resource key in `target_resource_params`. However there is an easier way: 

```suggestion
        base_cmpr_op = cls.resource_rep(**target_resource_params)
```